src/alto-ui/Form/TagInput/TagInput.js

```diff
@@ -0,0 +1,209 @@
+      const newTags = tagsToRemove.reduce((acc, tag) => onRemoveTag(tag, acc), tags);
+      if (newTags !== tags) {
+        onChange(newTags);
+      }
+    }
+    useSelectionKeyboardNav(state, handleRemoveTags);
+
+    const onNewTag = props.onNewTag || ((tag, add) => add(tag));
+
+    function handleNewTag(tag) {
+      const index = position === null || position === undefined ? tags.length : position;
+      const add = t => [...tags.slice(0, index), t, ...tags.slice(index)];
+      const newTags = onNewTag(tag, add, index, tags);
+      if (newTags !== tags) {
+        const focused = position !== undefined;
+        if (focused) {
+          resetValue();
+          if (position !== null && newTags.length > tags.length) {
+            setPosition(position + 1);
+          }
+        }
+        onChange(newTags);
+      }
+    }
+
+    function handleKeyDownInput(e) {
+      const val = value;
+      if (val) {
+        if (e.key === 'Enter' && match()) {
+          handleNewTag(val);
+        }
+        return undefined;
+      }
+      if (e.key === 'ArrowLeft') return movePositionLeft();
+      if (e.key === 'ArrowRight') return movePositionRight();
+      if (e.key === 'Backspace' && tags.length && position !== 0) {
+        const indexToRemove = position === null ? tags.length - 1 : position - 1;
+        const newTags = tags.filter((_, index) => index !== indexToRemove);
+        onChange(newTags);
+        if (position !== null) setPosition(indexToRemove);
+      }
+      return undefined;
+    }
+
+    function handleClickTag(e, tag, index) {
+      if (isTagActive(tag)) {
+        removeFromSelection(tag);
+      } else {
+        const tagsToAdd =
+          e.shiftKey && selection.length
+            ? sliceArr(tags, tags.findIndex(isTagEqual(selection.slice(-1)[0])), index)
+            : [tag];
+        addToSelection(...tagsToAdd);
+      }
+    }
+
+    const input = (
+      <input
+        ref={refs.input}
+        type="text"
+        className={bemClass('TagInput__input', {
+          empty: !value,
+          valid: match(),
+        })}
+        value={value}
+        onChange={e => setValue(e.target.value)}
+        size={value ? value.length : 0}
+        onFocus={resetSelection}
+        onBlur={() => {
+          resetValue();
+          resetPosition();
+        }}
+        onKeyDown={handleKeyDownInput}
+        disabled={disabled}
+      />
+    );
+
+    return (
+      <div ref={refs.main} className={bemClass('TagInput', {}, className)}>
+        <div className="TagInput__tags-list TagInput__tags-list--value" ref={refs.value}>
+          <button
+            disabled={disabled}
+            className="TagInput__button"
+            onClick={() => setPosition(null)}
+          />
+          {tags.map((tag, index) => (
+            <Fragment key={getTagValue(tag)}>
+              {position === index ? input : <div style={{ width: 1 }} />}
```

Comment:
`<div style={{ width: 1 }} />` maybe some class instead set here style ? `TagInput__carret` ?

Reply:
yep you' re right